### PR TITLE
Add IsEmpty and Length properties to System.BinaryData

### LINKED ISSUE #77970: [API Proposal]: Add Length and IsEmpty properties to System.BinaryData
### Background and motivation

Getting the length of a `System.BinaryData` instance is something quite natural. Currently, the best way to get the length is by doing `data.ToMemory().Length` which is not very pleasant.

There's also a trap where one could do `data.ToArray().Length` or even  `data.ToStream().Length` and this would allocate!

Having a `Length` property solves both issues: code is more readable and ensures no allocation.

An `IsEmpty` property would also be welcome.

### API Proposal

```csharp
namespace System;

public class BinaryData
{
    public int Length { get; }
    public bool IsEmpty { get; }
}
```

### API Usage

```csharp
void ProcessData(BinaryData data, string outputFile)
{
    if (data.IsEmpty)
    {
        Console.WriteLine($"No data available to write to {outputFile}");
    }
    else
    {
        Console.WriteLine($"Writing {data.Length} bytes to {outputFile}");
        // ... actual implementation ...
    }
}
```

### Alternative Designs

_No response_

### Risks

There are no risks associated with this proposition. The implementations of those properties are trivial and would just call respectively `_bytes.Length` and `_bytes.IsEmpty`.

### PR DESCRIPTION
**Pull request by 0xced**
Fixes #77970

### CODE REVIEW

File: src/libraries/System.Memory.Data/tests/BinaryDataTests.cs [hunk 1]
```diff
@@ -614,6 +614,32 @@ public void ToStringReturnEmptyStringWhenBinaryDataEmpty()
...
+        public void LengthReturnsNumberOfBytes(int count)
+        {
+            var data = BinaryData.FromBytes(new byte[count]);
+            Assert.Equal(count, data.Length);
+        }
+
+        [Fact]
+        public void BinaryDataEmptyIsEmpty()
+        {
+            Assert.True(BinaryData.Empty.IsEmpty);
```
Comment (danmoseley) [inline src/libraries/System.Memory.Data/tests/BinaryDataTests.cs:630]: I'd add tests that both `new BinaryData(new byte[] { }))` and `new BinaryData("").IsEmpty` are both IsEmpty and Length==0

and test Length and IsEmpty for all of byte[], string and ROM.

Comment (0xced) [reply src/libraries/System.Memory.Data/tests/BinaryDataTests.cs:650]: Done in c507b04e485284626bdd6794b1a6223f8a915c9c

File: src/libraries/System.Memory.Data/tests/BinaryDataTests.cs [hunk 2]
```diff
@@ -614,6 +614,93 @@ public void ToStringReturnEmptyStringWhenBinaryDataEmpty()
...
+            Assert.Equal(count, data.Length);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(4)]
+        [InlineData(7)]
+        public void LengthReturnsNumberOfBytesForBinaryDataFromString(int count)
+        {
+            var data = BinaryData.FromBytes(new byte[count]);
```
Comment (adamsitnik) [inline src/libraries/System.Memory.Data/tests/BinaryDataTests.cs:643]: the test name ends with `FromString` but you are using `FromBytes` here. I assume you meant using `FromString`?

```suggestion
            var data = BinaryData.FromString(new string('a', count));
```

**Review by adamsitnik — COMMENTED** [review]
Overall it LGTM, I found just one test issue. As soon as it's addressed, I am ready to merge.

@0xced thank you for your contribution!

File: src/libraries/System.Memory.Data/tests/BinaryDataTests.cs (hunk 2 above)
Comment (0xced) [reply src/libraries/System.Memory.Data/tests/BinaryDataTests.cs]: Absolutely! I just force-pushed 8f5328ae4ec.

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @0xced !

**Comment by danmoseley** [comment]
Thank you @0xced .. Perhaps you're interested in another contribution some place?

**Pull request merged**